agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileSender.java

```diff
@@ -56,12 +57,12 @@ private LocalFileSender(LocalFileLoader localFileLoader, TelemetryChannel teleme
 
   @Override
   public void run() {
-    // TODO (heya) should this loop more aggressively until it reads all telemetry from disk?
-    //  or is it intentional to only send one batch per 30 seconds?
```

Comment:
is this comment addressed now?

Reply:
yes. will delete that.

Reply:
what is the resolution? is it not needed to loop more aggressively, or is it now looping more aggressively?

Reply:
resultCode.join(30, TimeUnit.SECONDS); // wait max 30 seconds for request to be completed.

it's still sending every 30 seconds, but it's less aggressive.

Reply:
let's say a process can't connect to breeze for 1 day, that could be `86400 / 5 = 17280` files, which would take `17280 * 30 = 518400` seconds (or 6 days) to send them all at 1 file per 30 seconds?

Reply:
does batch has a capacity limit? we don't handle that case yet. i will add the todo back.